addons/hw_drivers/drivers/PrinterDriver.py

```diff
@@ -0,0 +1,299 @@
+    if device_id in iot_devices:
+        reason = reason if reason != 'none' else None
+        state_value = {
+            IPP_PRINTER_IDLE: 'connected',
+            IPP_PRINTER_PROCESSING: 'processing',
+            IPP_PRINTER_STOPPED: 'stopped'
+        }
+        iot_devices[device_id].update_status(state_value[state], message, reason)
+
+# Create a Cups subscription if it doesn't exist yet
+try:
+    conn.getSubscriptions('/printers/')
+except IPPError:
+    conn.createSubscription(
+        uri='/printers/',
+        recipient_uri='dbus://',
+        events=['printer-state-changed']
+    )
+
+# Listen for notifications from Cups
+os.environ['DISPLAY'] = ":0.0"
+bus = dbus.SystemBus()
+bus.add_signal_receiver(cups_notification_handler, signal_name="PrinterStateChanged", dbus_interface="org.cups.cupsd.Notifier")
+
+
+class PrinterDriver(Driver):
+    connection_type = 'printer'
+    status = {'status': "disconnected", 'printers': [], 'messages': []}
+
+    def __init__(self, device):
+        super(PrinterDriver, self).__init__(device)
+        self._device_type = 'printer'
+        self._device_connection = self.dev['device-class'].lower()
+        self._device_name = self.dev['device-make-and-model']
+        self._device_identifier = self.dev['identifier']
+        self.state = {
+            'status': 'connecting',
+            'message': 'Connecting to printer',
+            'reason': None,
+        }
+        self.send_status()
+        self.add_connected_printer()
+        if 'direct' in self._device_connection:
+            self.print_status()
+
+    @classmethod
+    def supported(cls, device):
+        protocol = ['dnssd', 'lpd']
+        if any(x in device['url'] for x in protocol) or 'direct' in device['device-class']:
+            model = cls.get_device_model(device)
+            ppdFile = ''
+            for ppd in PPDs:
+                if model and model in PPDs[ppd]['ppd-product']:
+                    ppdFile = ppd
+                    break
+            with cups_lock:
+                if ppdFile:
+                    conn.addPrinter(name=device['identifier'], ppdname=ppdFile, device=device['url'])
+                else:
+                    conn.addPrinter(name=device['identifier'], device=device['url'])
+                if device['identifier'] not in printers:
+                    conn.setPrinterInfo(device['identifier'], device['device-make-and-model'])
+                    conn.enablePrinter(device['identifier'])
+                    conn.acceptJobs(device['identifier'])
+                    conn.setPrinterUsersAllowed(device['identifier'], ['all'])
+                    conn.addPrinterOptionDefault(device['identifier'], "usb-no-reattach", "true")
+                    conn.addPrinterOptionDefault(device['identifier'], "usb-unidir", "true")
+                else:
+                    device['device-make-and-model'] = printers[device['identifier']]['printer-info']
+            if 'STR_T' in device['device-id']:
```

Comment:
deplace to line 100
because you return false if 'STR_T' in device['device-id'].
So not needed to check PPD file or add printer